commerce_coordinator/apps/commercetools/clients.py

```diff
@@ -369,6 +369,28 @@ def get_payment_by_transaction_interaction_id(self, interaction_id: str) -> CTPa
         logger.info(f"[CommercetoolsAPIClient] - Attempting to find payment with interaction ID {interaction_id}")
         return self.base_client.payments.query(where=f'transactions(interactionId="{interaction_id}")').results[0]
 
+    def get_product_variant_by_program(self, product_id: str) -> Optional[CTProductVariant]:
+        """
+        Fetches a program from Commercetools.
+        Args:
+            product_id: The ID of the program (bundle) to fetch.
+        Returns:
+            CTProductVariant if found, None otherwise.
+        """
+        start_time = datetime.datetime.now()
+        try:
+            product_projection = self.base_client.product_projections.get_by_key(product_id)
```

Comment:
just for my info. Why aren't we using.
```suggestion
            product_projection = self.base_client.products.get_by_key(product_id)
```